### PR TITLE
Add mount_mode to MOUNT_STATUS

### PR DESCRIPTION
**Pull request by hendjoshsr71**
Partner Request via Discord

This adds mount_mode to the MOUNT_STATUS message as an extension. 

Will send this for upstream MAVLINK merge once I get feedback.

### CODE REVIEW

File: message_definitions/v1.0/ardupilotmega.xml [hunk 1]
```diff
@@ -1210,6 +1210,8 @@
       <field type="int32_t" name="pointing_a" units="cdeg">Pitch.</field>
       <field type="int32_t" name="pointing_b" units="cdeg">Roll.</field>
       <field type="int32_t" name="pointing_c" units="cdeg">Yaw.</field>
+      <extensions/>
+      <field name="mount_mode" type="uint8_t" enum="MAV_MOUNT_MODE">Mount operating mode.</field>
```
Comment (peterbarker) [inline message_definitions/v1.0/ardupilotmega.xml:1214]:
```suggestion
      <field type="uint8_t" name="mount_mode" enum="MAV_MOUNT_MODE">Mount operating mode.</field>
```

**Review by peterbarker — CHANGES_REQUESTED** [review]
The mode isn't enough to know everything about what the mode is doing.

 Bits we still won't know:
 - whether it is stabilising on rpy
 - where it is pointing for ROI / wpnext / mavlink system ID

Still... this won't preclude adding more things, I guess...

**Comment by hendjoshsr71** [comment]
> 

Agree! Though if we start adding those I wonder if they are better sent out as an additional message instead of with mode_status? Since they will likely not change as often?

I'm happy to add more fields too but need guidance from users of the message and gimbals.

**Review by tridge — APPROVED** [review]

**Review by peterbarker — APPROVED** [review]

**Comment by hendjoshsr71** [comment]
Created a companion PR for this to the upstream MAVLINK:  https://github.com/mavlink/mavlink/pull/1777

**Pull request merged**